src/components/Button.tsx

```diff
@@ -12,9 +19,13 @@ interface ButtonProps extends DefaultProps, Coloring, Sizing {
 	iconElement?: JSX.Element;
 }
 
+interface ThemedButtonProps extends ButtonProps {
```

Comment:
We typically prefer interfaces over types, is there a reason to use a type here?

Reply:
Totally with you on interface preference (with the exception of empty interfaces that only inherit from others, where tslint prefers `type`s).

Actually I was just saying that we could use `ThemedStyledProps<ButtonProps, Theme>` or `ThemeProps<Theme>` (even in the interface) instead of defining the `theme` prop ourselves. Perhaps some of the built in styled-component methods could have better inference if we do so. No strong opinions though.

Reply:
Ah I see what you mean, yep sounds good

Reply:
I guess we can keep this as is. Was NIT anyway, although the code would be DRYer even if we had such an interface in our `utils.ts`.